src/main/java/eu/okaeri/validator/OkaeriValidator.java

```diff
@@ -93,9 +101,79 @@ public Set<ConstraintViolation> validateParameters(@NonNull Parameter[] paramete
 
     @Override
     public Set<ConstraintViolation> validateParameter(@NonNull Parameter parameter, Object value) {
-        return this.validationProviders.values().stream()
+        Set<ConstraintViolation> violations = this.validationProviders.values().stream()
             .filter(provider -> provider.shouldValidate(parameter))
             .flatMap(provider -> provider.validate(parameter, value).stream())
             .collect(Collectors.toCollection(LinkedHashSet::new));
+        violations.addAll(this.validate(parameter.getName(), parameter.getAnnotatedType(), value));
+        return violations;
+    }
+
+    private Set<ConstraintViolation> validate(@NonNull String fieldName, @NonNull AnnotatedType annotatedObjectType, Object object) {
+        if (!(annotatedObjectType instanceof AnnotatedParameterizedType)) {
+            return new LinkedHashSet<>();
+        }
+        AnnotatedParameterizedType parameterizedType = (AnnotatedParameterizedType) annotatedObjectType;
+
+        if (!(parameterizedType.getType() instanceof ParameterizedType)) {
+            return new LinkedHashSet<>();
+        }
+        Class<?> objectType = (Class<?>) ((ParameterizedType) parameterizedType.getType()).getRawType();
+
+        AnnotatedType[] actualTypes = parameterizedType.getAnnotatedActualTypeArguments();
+        if (Collection.class.isAssignableFrom(objectType)) {
+            return this.validateCollection(fieldName, actualTypes[0], (Collection<?>) object);
+        } else if (Map.class.isAssignableFrom(objectType)) {
+            return this.validateMap(fieldName, actualTypes[0], actualTypes[1], (Map<?, ?>) object);
+        }
+        return new LinkedHashSet<>();
+    }
+
+    @Override
+    public Set<ConstraintViolation> validateCollection(@NonNull String fieldName, @NonNull AnnotatedType annotatedValueType, @NonNull Collection<?> collection) {
+        Type valueType = annotatedValueType.getType();
+        if (!(valueType instanceof Class<?>)) {
+            return new LinkedHashSet<>();
+        }
+
+        AtomicInteger index = new AtomicInteger(0);
+        return collection.stream()
+            .flatMap(element -> this.validateElement(fieldName, (Class<?>) valueType, annotatedValueType, element).stream()
+                .peek(elementViolation -> elementViolation.setField(fieldName + " [element index " + index.getAndIncrement() + "]")))
```

Comment:
I think this needs some new field in ConstraintViolation. Similarly to propertyPath in Jakarta EE:

ConstraintViolation:
- String field
- **String path**
- String message
- String type

For now, nullable String would do:
- collection element validation should produce path: "[index]" e.g. `[0]` `[1]`
- map key validation should produce path: "[key]" e.g. `[aaa]` `[bbbb]`
- map value validation should produce path: "[key].value" e.g. `[aaa].value` `[bbbb].value`

This should be consistent with Jakarta EE string representation of Path used in theirs ConstraintViolation.